### ISSUE TITLE
A cluster with one node and no device "Volume create fails"

### ISSUE THREAD

**Issue by MohamedAshiqrh** [heketi/heketi#730]
There are multiple cluster in heketi but first cluster was having only one node and no devices.
volume create failed with:
`Error calling v.allocBricksInCluster: Id not found`

```
➜  heketi git:(testq) ✗ heketi-cli topology load --json=topology-sample.json            
Creating cluster ... ID: d3255536ffb1ac6122d004157017fe92
        Creating node 192.168.10.100 ... ID: 736dd4f9d1512d98dd112de22912a782
➜  heketi git:(testq) ✗ heketi-cli volume create --size=1
Error: Error calling v.allocBricksInCluster: Id not found
```
LOGS:
```
[negroni] Started POST /volumes
[heketi] DEBUG 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry.go:84: [cff688212f2fcfcb246f6cfd3abab88e] Replica 3
[negroni] Completed 202 Accepted in 189.541µs
[asynchttp] INFO 2017/03/28 13:50:47 Started job 8999d43edd9e6b8d4330a88da71d31c3
[heketi] INFO 2017/03/28 13:50:47 Creating volume cff688212f2fcfcb246f6cfd3abab88e
[heketi] DEBUG 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry.go:250: Using the following clusters: [d3255536ffb1ac6122d004157017fe92]
[heketi] DEBUG 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry_allocate.go:44: brick_size = 1048576
[heketi] DEBUG 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry_allocate.go:45: sets = 1
[negroni] Started GET /queue/8999d43edd9e6b8d4330a88da71d31c3
[heketi] DEBUG 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry_allocate.go:46: num_bricks = 3
[heketi] INFO 2017/03/28 13:50:47 brick_num: 0
[heketi] ERROR 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/allocator_simple.go:157: Unknown cluster id requested: d3255536ffb1ac6122d004157017fe92
[negroni] Completed 200 OK in 17.69µs
[heketi] DEBUG 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry_allocate.go:339: 0 / 3
[heketi] DEBUG 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry_allocate.go:306: Error detected.  Cleaning up volume cff688212f2fcfcb246f6cfd3abab88e: Len(0)
[heketi] ERROR 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry_allocate.go:61: Id not found
[heketi] ERROR 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/volume_entry.go:308: Error calling v.allocBricksInCluster: Id not found
[heketi] ERROR 2017/03/28 13:50:47 /src/github.com/heketi/heketi/apps/glusterfs/app_volume.go:148: Failed to create volume: Error calling v.allocBricksInCluster: Id not found
[asynchttp] INFO 2017/03/28 13:50:47 Completed job 8999d43edd9e6b8d4330a88da71d31c3 in 5.919237ms
[negroni] Started GET /queue/8999d43edd9e6b8d4330a88da71d31c3
[negroni] Completed 500 Internal Server Error in 31.321µs
```

**Comment by MohamedAshiqrh** [comment]
@obnoxxx This for the issue which we saw.

**Comment by humblec** [comment]
Yeah, this error need to be more meaningful @MohamedAshiqrh

**Comment by MohamedAshiqrh** [comment]
This is same as https://github.com/heketi/heketi/issues/712

**Comment by obnoxxx** [comment]
@purpleidea you might want to look into this one.

**Comment by nixpanic** [comment]
@obnoxxx, I created a cluster with a singe storage node without bricks. The error is different now:
```
[root@localhost ~]# ./heketi-cli volume create --size=2
Error: No space
```

I guess this can be closed, similar to #712.

**Comment by phlogistonjohn** [comment]
As noted the behavior in this area has significantly changed in the last two releases. Closing issue. Please reopen if additional discussion is needed.
